### PR TITLE
fix(filter): make anthropic_stream_events self-arming via metadata and Content-Type

### PR DESCRIPTION
**Pull request by leseb**
## Summary

- Replace the default-open negative gate (`is_known_non_streaming_transform`) with positive arming: the filter now checks upstream streaming metadata (`anthropic_messages_format.stream` or `anthropic_to_openai.streaming`) AND validates the response `Content-Type` is `text/event-stream` (media-type-aware, handles `charset` parameters like `text/event-stream; charset=utf-8`) before arming.
- This makes the filter self-contained — `response_conditions` in YAML config become an optional optimization rather than a correctness requirement.
- Add 6 new unit tests covering arming logic (positive and negative cases, both metadata sources, charset parameter handling, body passthrough when unarmed).

## Test plan

- [x] `make lint` passes (clippy + fmt + xtask checks)
- [x] `make build` passes
- [x] Unit tests pass (`cargo test -p praxis-filter`)
- [x] Integration tests pass (`cargo test -p praxis-tests-integration --test suite -- anthropic`)
- [ ] CI passes all checks

### CODE REVIEW

File: filter/src/builtins/http/ai/anthropic/stream_events/mod.rs [hunk 1]
```diff
@@ -649,11 +754,51 @@ mod tests {
...
+        let mut body = Some(Bytes::from(chunk));
+        drop(filter.on_response_body(&mut ctx, &mut body, false).unwrap());
+
+        let out = String::from_utf8(body.unwrap().to_vec()).unwrap();
+        assert!(
+            !out.contains("message_start"),
+            "unarmed filter should not transform body"
+        );
+        assert!(
+            out.contains("gpt-4"),
```
[the remark this answers is not in the document]
Comment (leseb) [reply filter/src/builtins/http/ai/anthropic/stream_events/mod.rs]: 6e8da06196cef37bb07945fe2a1d2c61c8aab064

File: filter/src/builtins/http/ai/anthropic/stream_events/mod.rs [hunk 2]
```diff
@@ -183,19 +186,54 @@ fn process_sse_chunk(ctx: &mut HttpFilterContext<'_>, chunk_str: &str) -> Bytes
...
+            .filter_metadata
+            .get("anthropic_to_openai.streaming")
+            .is_some_and(|v| v == "true")
+}
+
+/// Whether a `Content-Type` header value indicates `text/event-stream`.
+fn is_event_stream_content_type(ct: &str) -> bool {
+    ct.split(';')
+        .next()
+        .is_some_and(|media| media.trim().eq_ignore_ascii_case("text/event-stream"))
```
[the remark this answers is not in the document]
Comment (leseb) [reply filter/src/builtins/http/ai/anthropic/stream_events/mod.rs:402]: 6e8da06196cef37bb07945fe2a1d2c61c8aab064
